feat(ONB-204): add keyword counter page with frequency analysis

nagmex210 opened the pull request:
- Create VKeywordCounter page with side-by-side responsive layout
  - Implement real-time word count display as user types
  - Add keyword frequency analysis with count and percentage calculation
  - Include responsive TailwindCSS table for results display
  - Add chart-bar icon to card header and navigation tab
  - Support static text prop for initial load from parent pages

nagmex210 committed:
feat(ONB-204): add keyword counter page with frequency analysis

  - Create VKeywordCounter page with side-by-side responsive layout
  - Implement real-time word count display as user types
  - Add keyword frequency analysis with count and percentage calculation
  - Include responsive TailwindCSS table for results display
  - Add chart-bar icon to card header and navigation tab
  - Support static text prop for initial load from parent pages

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+const keywordStats = ref<{ keyword: string; count: number; percentage: number }[]>([]);
+const totalWords = ref(0);
+
+const processText = (text: string) => {
```
When I read your code i should understand what is the intend by reading function name. It should be specific for that, It could be long tho. So, we should change function name with better alternative like `textToWords` kinda

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+  }
+});
+
+watch(inputText, updateWordCount);
```
Please don't use watcher it's a not good practice

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+};
+
+onMounted(() => {
+  if (props.text) {
+    inputText.value = props.text;
+    updateWordCount();
+  }
```
I don't recommend inline logic inside life cycle events. That increase complexity in the future. Please move this logic inside a function instead.

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+
+watch(inputText, updateWordCount);
+
+const countWords = (words: string[]) => {
+  const wordCount: { [key: string]: number } = {};
+  words.forEach(word => {
+    wordCount[word] = (wordCount[word] || 0) + 1;
+  });
+  return wordCount;
+};
```
This could be computed we don't need to do this manually

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+  return wordCount;
+};
+
+const calculatePercentage = (count: number, total: number) => {
+  return Math.round((count / total) * 100 * 100) / 100;
+};
```
This also can be computed property

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+  return Math.round((count / total) * 100 * 100) / 100;
+};
+
+const formatKeywordStats = (wordCount: { [key: string]: number }, totalWords: number) => {
+  return Object.entries(wordCount)
+    .map(([keyword, count]) => ({
+      keyword,
+      count,
+      percentage: calculatePercentage(count, totalWords)
+    }))
+    .sort((a, b) => b.count - a.count);
+};
```
This one also could be computed property

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,274 @@
+              class="mb-4"
+            />
+            
+            <ma-button 
+              type="primary" 
+              size="medium"
+              @click="analyzeKeywords"
+              class="w-full sm:w-auto"
+            >
```
Events should be last
```suggestion
            <ma-button 
              type="primary" 
              size="medium"
              class="w-full sm:w-auto"
              @click="analyzeKeywords"
            >
```

nagmex210 committed:
refactor(keyword-counter): improve code quality and reactivity

  - Rename processText to textToWords for better clarity
  - Replace watcher with computed properties for automatic reactivity
  - Extract inline lifecycle logic to initializeText function
  - Convert manual functions to computed properties:
    - countWords → wordCount computed using reduce()
    - keywordStats → computed with integrated percentage calculation
  - Remove analyzeKeywords function and button for real-time updates
  - Remove calculatePercentage and formatKeywordStats helper functions
  - Update UI text to reflect automatic behavior

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,248 @@
+  }
+};
+
+onMounted(initializeText);
```
```suggestion
onMounted(() => {
	initializeText()
});
```

When you have multiple operations you can chain them that way

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,248 @@
+
+onMounted(initializeText);
+
+const hasResults = computed(() => keywordStats.value.length > 0);
```
haveResults or hasResult

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,248 @@
+        </ma-card>
+      </div>
+
+      <!-- Results Table Section -->
```
```suggestion
```

turanibrahim commented on src/pages/VKeywordCounter.vue:
```diff
@@ -0,0 +1,248 @@
+<template>
+  <div class="keyword-counter-wrapper">
+    <div class="main-container">
+      <!-- Input Section -->
```
```suggestion
```

turanibrahim reviewed (changes requested):
We are so close to end. Please update the unit test I would like to have good coverage. Can you add a coverage library to see coverage?
